daemon/logger/journald/read.go

```diff
@@ -170,13 +170,14 @@ func (s *journald) Close() error {
 	return nil
 }
 
-func (s *journald) drainJournal(logWatcher *logger.LogWatcher, config logger.ReadConfig, j *C.sd_journal, oldCursor *C.char) *C.char {
+func (s *journald) drainJournal(logWatcher *logger.LogWatcher, config logger.ReadConfig, j *C.sd_journal, oldCursor *C.char, untilUnixMicro uint64) *C.char {
```

Comment:
Given that we already pass `logger.ReadConfig`, why not add `until` to that config? https://github.com/moby/moby/blob/27bd6842f8518780b99fdb42f8e7f84c17856d87/daemon/logger/logger.go#L83-L88

```go
// ReadConfig is the configuration passed into ReadLogs.
type ReadConfig struct {
	Since  time.Time
	Tail   int
	Follow bool
}
```

edit: I see you already added `Until` to that type as part of this PR